src/validation/rules/FieldsOnCorrectType.js

```diff
@@ -11,14 +11,70 @@
 import type { ValidationContext } from '../index';
 import { GraphQLError } from '../../error';
 import type { Field } from '../../language/ast';
-import type { GraphQLType } from '../../type/definition';
-
+import {
+  isAbstractType,
+  GraphQLAbstractType,
+  GraphQLObjectType,
+} from '../../type/definition';
 
 export function undefinedFieldMessage(
-  fieldName: string,
-  type: GraphQLType
-): string {
-  return `Cannot query field "${fieldName}" on "${type}".`;
+    field: string,
+    type: string,
+    suggestedTypes: Array<string>): string {
```

Comment:
minor style nit: the rest of the codebase drops the return type of >80ch functions to it's own line.

e.g.

```
export function undefinedFieldMessage(
  field: string,
  type: string,
  suggestedTypes: Array<string>
): string {